src/main/java/spoon/refactoring/RenameLocalVariableRefactor.java

```diff
@@ -0,0 +1,266 @@
+/**
+ * Copyright (C) 2006-2017 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.refactoring;
+
+import java.util.Collection;
+import java.util.regex.Pattern;
+
+import spoon.SpoonException;
+import spoon.reflect.code.CtCatchVariable;
+import spoon.reflect.code.CtLocalVariable;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtField;
+import spoon.reflect.declaration.CtParameter;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.CtVariable;
+import spoon.reflect.reference.CtFieldReference;
+import spoon.reflect.reference.CtLocalVariableReference;
+import spoon.reflect.reference.CtReference;
+import spoon.reflect.reference.CtVariableReference;
+import spoon.reflect.visitor.Filter;
+import spoon.reflect.visitor.chain.CtConsumer;
+import spoon.reflect.visitor.chain.CtQueryable;
+import spoon.reflect.visitor.chain.CtScannerListener;
+import spoon.reflect.visitor.chain.ScanningMode;
+import spoon.reflect.visitor.filter.LocalVariableReferenceFunction;
+import spoon.reflect.visitor.filter.LocalVariableScopeFunction;
+import spoon.reflect.visitor.filter.PotentialVariableDeclarationFunction;
+import spoon.reflect.visitor.filter.SiblingsFunction;
+import spoon.reflect.visitor.filter.SiblingsFunction.Mode;
+import spoon.reflect.visitor.filter.VariableReferenceFunction;
+
+/**
+ * Spoon model refactoring function which renames `target` local variable to `newName`<br>
+ * This refactoring will throw {@link RefactoringException} if the model would be not consistent after rename to new name.
+ * The exception is thrown before the model modificatons are started.
+ * <pre>
+ * CtLocalVariable anLocalVariable = ...
+ * RenameLocalVariableRefactor refactor = new RenameLocalVariableRefactor();
+ * refactor.setTarget(anLocalVariable);
+ * refactor.setNewName("someNewName");
+ * try {
+ *   refactor.refactor();
+ * } catch (RefactoringException e) {
+ *   //handle name conflict or name shadowing problem
+ * }
+ * </pre>
+ */
+public class RenameLocalVariableRefactor extends AbstractRenameRefactor<CtLocalVariable<?>> {

```

Comment:
add a static method in Refactoring (eg `Refactoring#renameLocalVariable(CtNamedElement target, String newName)`) to help discoverability and usage of this class?